src/window.c

```diff
@@ -190,6 +191,58 @@ int S2D_Show(S2D_Window *window) {
             window->on_mouse(event);
           }
           break;
+        case SDL_FINGERMOTION:
+         {
+            if (window->on_touch) {
+           float fx,dx, fy, dy; 
+           fx = e.tfinger.x * window->width; // TODO: Viewport in scaling
+           fy = e.tfinger.y * window->height;
+           dx = e.tfinger.dx * window->width;
+           dy = e.tfinger.dy * window->height;
+           S2D_GetMouseOnViewport(window, (int)fx, (int)fy, &mx, &my);
+            S2D_Event event = {
+              .x = mx, .y = my, .delta_x = dx, .delta_y = dy, .finger_id = e.tfinger.fingerId,
+              .touch_id = e.tfinger.touchId
+            };
+            event.type = S2D_FINGER_MOTION;
+            window->on_touch(event);
+          }
+
+        }break;
+	case SDL_FINGERDOWN:
```

Comment:
Since `SDL_FINGERDOWN` and `SDL_FINGERUP` are identical, save for `event.type`, I might suggest combining both events like in [mouse down/up here](https://github.com/simple2d/simple2d/blob/v0.6.0/src/window.c#L161). (Reminds me, we can probably do the same for `SDL_JOYBUTTONDOWN` and `SDL_JOYBUTTONUP` as well.)